### PR TITLE
Completed car dealership calculator

### PR DESCRIPTION
**Pull request by kyshang**
Completed car dealership calculator project. Edited customers.py, vehicles.py, contracts.py.

### CODE REVIEW

**Comment by santiagobasulto** [comment]
This PR will be reviewed by: @erikasboren

File: dealership/contracts.py [hunk 1]
```diff
@@ -1,12 +1,49 @@
...
 class BuyContract(Contract):
     def __init__(self, vehicle, customer, monthly_payments):
-        pass
-
+        self.vehicle = vehicle
+        self.customer = customer
+        self.monthly_payments = monthly_payments
+        
+    def interest_rate(self):
+        interest_dict = {Car : 1.07, Motorcycle: 1.03, Truck : 1.11}
```
Comment (erika-boren) [inline dealership/contracts.py:13]: I like the use of the dictionary for the interest and lease multipliers. Another way I saw a few others do implement it is buy adding a constant variable to the car, truck, and motorcycle class and calling it using self.vehicle.interest.

File: dealership/vehicles.py [hunk 2]
```diff
@@ -1,15 +1,30 @@
...
+    def sale_price(self):
+        return self.base_price * self.purchase_multiplier
+    
+    def purchase_price(self):
+        return self.sale_price() - (self.sale_multiplier * self.miles)
 
 class Car(Vehicle):
-    pass
-
+    purchase_multiplier = 1.2
```
Comment (erika-boren) [inline dealership/vehicles.py:19]: I like how you implemented this, well done. I refactored mine the same way :)

File: dealership/contracts.py [hunk 3]
```diff
@@ -1,12 +1,49 @@
...
+        return (interest_dict[type(self.vehicle)] * self.monthly_payments * self.vehicle.sale_price() / 100)
+    
+    def total_value(self):
+        price = self.vehicle.sale_price() + self.interest_rate() 
+        if not self.customer.is_employee():
+            return price
+        else:
```
Comment (erika-boren) [inline dealership/contracts.py:20]: You could eliminate these 3 lines of code with just "return price - (price * 0.1) 

 else:
             discount = price * 0.1
             return price - discount

**Review by erika-boren — COMMENTED** [review]
Great job on the assignment. I thought your code was written very elegantly. I learned a few things and refactored my code the same way. I made a few notes on things I saw implemented differently, just as a suggestion. Great job!

**Pull request closed without merging**